File: assignments/src/main/java/vendingmachine/VendingMachine.java
package vendingmachine;

import java.util.*;

public class VendingMachine {
    private Scanner sc = new Scanner(System.in);

    private static final int COKE = 500, CIDER = 700, FANTA = 300, WATER = 200;
    private int money;

    private void printMenu() {
        System.out.println("========================================");
        System.out.println("              자판기 메뉴");
        System.out.println("[1] 콜라   - 500원");
        System.out.println("[2] 사이다 - 700원");
        System.out.println("[3] 환타   - 300원");
        System.out.println("[4] 물     - 200원");
        System.out.println("[5] 돈 넣기");
        System.out.println("[6] 종료");
        System.out.println("========================================");
        System.out.println("현재 금액 : " + money + "원");
        System.out.println("========================================");
    }

    private void buyDrink(int price, String drinkName) {
        System.out.println("----------------------------------------");
        System.out.printf("%s 선택\n", drinkName);

        if (money >= price) {
            money -= price;
            System.out.println("구매 완료");
            System.out.println("남은 금액 : " + money + "원");
            System.out.println("----------------------------------------");
        } else {
            System.out.println("잔액 부족");
            System.out.println("----------------------------------------");
        }
    }

    private void addMoney(int add) {
        money += add;
        System.out.println("----------------------------------------");
        System.out.println("금액 추가 완료");
        System.out.println("현재 금액 : " + money + "원");
        System.out.println("----------------------------------------");
    }

    public void run() {
        System.out.println("----------------------------------------");
        System.out.print("금액 입력 : ");
        money = Integer.parseInt(sc.nextLine());

        while (true) {
            printMenu();

            System.out.println("----------------------------------------");
            System.out.print("메뉴 입력: ");
            int menu = Integer.parseInt(sc.nextLine());

            switch (menu) {
                case 1:
                    buyDrink(COKE, "콜라");
                    break;
                case 2:
                    buyDrink(CIDER, "사이다");
                    break;
                case 3:
                    buyDrink(FANTA, "환타");
                    break;
                case 4:
                    buyDrink(WATER, "물");
                    break;
                case 5:
                    System.out.println("----------------------------------------");
                    System.out.print("추가 금액 입력 : ");
                    int add = Integer.parseInt(sc.nextLine());
                    addMoney(add);
                    break;
                case 6:
                    System.out.println("========================================");
                    System.out.println("              자판기 종료");
                    System.out.println("========================================");
                    sc.close();
                    return;
                default:
                    System.out.println("----------------------------------------");
                    System.out.println("               !다시 입력!");
            }
        }
    }

    public static void main(String[] args) {
        VendingMachine vm = new VendingMachine();
        vm.run();
    }
}

File: assignments/kotlin/src/main/kotlin/vendingmachine/VendingMachine.kt
package vendingmachine

const val COKE = 500
const val CIDER = 500
const val FANTA = 300
const val WATER = 200

fun printMenu(totalMoney: Int) {
    println("================ 자판기 =================")
    println("[1]콜라: ${COKE}, [2]사이다: ${CIDER}, [3]환타: ${FANTA}")
    println("[4]물: ${WATER},   [5]돈 넣기,       [6]종료")
    println("----------------------------------------")
    println("현재 금액: $totalMoney")
    println("========================================")
}

fun getChoice(): Int {
    println("메뉴 선택: ")

    return readln().toInt()
}

fun getMoney(): Int {
    println("----------------------------------------")
    println("금액 투입: ")

    return readln().toInt()
}

fun calcMoney(totalMoney: Int, price: Int): Int {
    return totalMoney - price
}

fun calcMoneyException() {
    println("----------------------------------------")
    println("잔돈 부족")
}

fun main() {
    var totalMoney = 0

    while (true) {
        printMenu(totalMoney)
        val choice = getChoice()
        var result = -1

        when (choice) {
            1 -> {
                result = calcMoney(totalMoney, COKE)

                if (result < 0) {
                    calcMoneyException()
                }
                else {
                    totalMoney = result
                    println("----------------------------------------")
                    println("콜라 나옴")
                }
            }
            2 -> {
                result = calcMoney(totalMoney, CIDER)

                if (result < 0) {
                    calcMoneyException()
                }
                else {
                    totalMoney = result
                    println("----------------------------------------")
                    println("사이다 나옴")
                }
            }
            3 -> {
                result = calcMoney(totalMoney, FANTA)

                if (result < 0) {
                    calcMoneyException()
                }
                else {
                    totalMoney = result
                    println("----------------------------------------")
                    println("환타 나옴")
                }
            }
            4 -> {
                result = calcMoney(totalMoney, WATER)

                if (result < 0) {
                    calcMoneyException()
                }
                else {
                    totalMoney = result
                    println("----------------------------------------")
                    println("물 나옴")
                }
            }
            5 -> totalMoney += getMoney()
            6 -> {
                println("----------------------------------------")
                println("잔돈 ${totalMoney}원 반환")
                println("========================================")

                return
            }
            else -> {
                println("----------------------------------------")
                println("잘못된 입력. 다시 입력!")
            }
        }
    }
}